node/resolvers/saveMasterdata.ts

```diff
@@ -10,12 +10,21 @@ export async function saveMasterdata(
 
   sendMessageSplunk(saveValues.saveData.type, saveValues.saveData.content, ctx)
 
-  return ctx.clients.badges.save({
-    content: saveValues.saveData.content,
-    name: saveValues.saveData.name,
-    operator: saveValues.saveData.operator,
-    priority: saveValues.saveData.priority,
-    simpleStatements: saveValues.saveData.simpleStatements,
-    type: saveValues.saveData.type,
-  })
+  return ctx.clients.badges
+    .save({
+      content: saveValues.saveData.content,
+      name: saveValues.saveData.name,
+      operator: saveValues.saveData.operator,
+      priority: saveValues.saveData.priority,
+      simpleStatements: saveValues.saveData.simpleStatements,
+      type: saveValues.saveData.type,
+    })
+    .then(() => true)
+    .catch((e: any) => {
+      ctx.vtex.logger.error(
+        `Error to save Bagde through MasterdataV2 ${e.name} ${e.message}`
+      )
```

Comment:
Just a suggestion here. You could put `e.name` in another property, so you can filter your Splunk logs better. Example:

```suggestion
      ctx.vtex.logger.error(
        {
          message: 'Error to save badge through MasterdataV2',
          name: e.name,
          exception: e.message,
        }
      )
```

This way you could filter all "Error to save badge" errors on Splunk to see if they're all the same problem or different issues.